private/projects/cdk-graph-plugin-diagram.ts

```diff
@@ -68,11 +66,28 @@ export class CdkGraphPluginDiagramProject extends CdkGraphPluginProject {
     // Ensure sharp has cross-platform prebuilds included in bundled dependency
     // https://sharp.pixelplumbing.com/install#cross-platform
     const sharpPrebuildTask = this.addTask("sharp:prebuild", {
-      condition: '[ -n "$CI" ]',
       exec: "ts-node ./scripts/sharp-prebuild.ts",
     });
-
     this.packageTask.prependSpawn(sharpPrebuildTask);
+    // Ensure build input + output includes `sharp:prebuild` artifacts
+    this.nxOverride(
+      "targets.build.inputs",
+      [
+        "default",
+        "^default",
+        {
+          // To ensure sharp:prebuild artifacts are included in build cache hash
+          runtime:
+            "npx -p @aws-prototyping-sdk/nx-monorepo nx-dir-hasher {workspaceRoot}/packages/cdk-graph-plugin-diagram/node_modules/sharp",
+        },
+      ],
+      true
+    );
+    this.nxOverride(
+      "targets.build.outputs",
+      ["{projectRoot}/node_modules/sharp"],
+      true
+    );
```

Comment:
This is the core fix to the "sharp prebuild" issue, the rest is to properly support this.
- To update default "inputs" we needed to define "default" namedInput
- Nx wil **always** ignore all node_modules/* path, so had the create custom `nx-dir-hasher` supports hashing paths within node_modules
- Needed to remove CI conditional for prebuild so that local development can benefit from remote caching